extensions/windows-event-log/wel/MetadataWalker.cpp

```diff
@@ -17,21 +19,62 @@ bool MetadataWalker::for_each(pugi::xml_node &node) {
 
 			if (std::regex_match(attr.name(), regex_)) {
 				std::function<std::string(const std::string &)> idUpdate = [&](const std::string &input) -> std::string {
-					return resolve_ ? utils::OsUtils::userIdToUsername(input) : input;
+					if (resolve_) {
+						auto resolved = utils::OsUtils::userIdToUsername(input);
+						replaced_identifiers_[input] = resolved;
+						return resolved;
+					}
+					else {
+						replaced_identifiers_[input] = input;
+					}
 				};
 				updateText(node, attr.name(), std::move(idUpdate));
 
 			}
+
 			if (std::regex_match(attr.value(), regex_)) {
 				std::function<std::string(const std::string &)> idUpdate = [&](const std::string &input) -> std::string {
```

Comment:
We have exactly the same wrong function here. 
We should avoid moving and only create it once.

Reply:
@am-c-p-p this too. thanks @arpadboda